biogtr/io/association_matrix.py

```diff
@@ -102,61 +114,97 @@ def to_dataframe(
+                ]
+
+            else:
+                col_inds = np.arange(len(self.ref_instances))
 
         asso_df = pd.DataFrame(matrix, index=row_inds, columns=col_inds)
 
         return asso_df
 
     def reduce(
         self,
-        to: tuple[str] = ("inst", "traj"),
-        by: tuple[str] = (None, "pred"),
+        row_dims: str = "inst",
+        col_dims: str = "traj",
+        row_grouping: str = None,
+        col_grouping: str = "pred",
         reduce_method: callable = np.sum,
     ) -> pd.DataFrame:
-        """Reduce association matrix rows/columns to inst/traj x traj/inst.
+        """Aggregate the association matrix by specified dimensions and grouping.
 
         Args:
-           to: A tuple indicating how to reduce rows/columns. Either inst (remains unchanged), or traj (reduces matrix from n_query/n_ref to n_traj
-           by: A tuple indicating how to group rows/columns when aggregating. Either "pred" or "gt"
+           row_dims: A str indicating how to what dimensions to reduce rows to. Either inst (remains unchanged), or traj (n_rows=n_traj)
+           col_dims: A str indicating how to dimensions to reduce rows to. Either inst (remains unchanged), or traj (n_cols=n_traj)
+           row_grouping: A str indicating how to group rows when aggregating. Either "pred" or "gt".
+           col_grouping: A str indicating how to group columns when aggregating. Either "pred" or "gt".
            method: A callable function that operates on numpy matrices and can take an `axis` arg for reducing.
 
         Returns:
             The association matrix reduced to an inst/traj x traj/inst association matrix as a dataframe.
         """
-        row_to, col_to = to
-        row_by, col_by = by
-
-        if row_to not in ("inst", "traj") or col_to not in ("inst", "traj"):
+        if (
+            row_dims is not None
+            and row_dims.lower()
+            not in [key for keys in self.AVAILABLE_REDUCTIONS.values() for key in keys]
+        ) or (
+            col_dims is not None
+            and col_dims.lower()
+            not in [key for keys in self.AVAILABLE_REDUCTIONS.values() for key in keys]
+        ):
             raise ValueError(
-                f"Can only reduce to inst/traj x inst/traj but ({row_to} x {col_to}) was requested!"
+                f"Can only reduce to inst/traj x inst/traj but ({row_dims} x {col_dims}) was requested!"
             )
 
-        if row_by not in ("pred", "gt", None) or col_by not in ("pred", "gt", None):
+        if (
+            row_grouping not in self.AVAILABLE_GROUPINGS
+            or col_grouping not in self.AVAILABLE_GROUPINGS
+        ):
             raise ValueError(
-                f"Can aggregate by [gt, pred, None] but {row_by} and {col_by} was requested!"
+                f"Can aggregate by [gt, pred, None] but {row_grouping} and {col_grouping} was requested!"
             )
```

Comment:
Do we really need all this error checking? Again, keep the complexity in mind.

A list membership check is O(n), which is small for a small list, but in our case we'll be running this hundreds of thousands to millions of times, so every one of these checks adds up quick.

Refs:
- https://stackoverflow.com/a/13884213/1939934
- https://wiki.python.org/moin/TimeComplexity